Bms (#40)
LibreSolar/learn.libre.solar: docs/system/bms.md

# Battery Management System

> A Battery Management System (BMS) is an electronic circuit that can manage a rechargeable device such as a battery pack or a single cell. Like most electonics, accumulator are limited in the voltage and current they can handle. Furthermore it is vital for most types of rechargeable batteries to monitor discharge and charge cycles to ensure a long lifetime.

Reviewer: It should be made clear that a BMS is normally only needed for Li-ion batteries.

Also can you start with a distinction between cell, module and pack?

Literature:
https://everlasting-project.eu/wp-content/uploads/2016/11/EVERLASTING_D6.1_final_20170228.pdf
https://nhresearch.com/wp-content/uploads/app-note-fundamentals-battery-module-pack-test.pdf
https://www.samsungsdi.com/column/all/detail/54344.html
https://warwick.ac.uk/fac/sci/wmg/business/automotive_batteries_101_wmg-apc.pdf

## Main Functions

### Electric Cell Protection

As described in chapter ["Battery"](battery.md), different cells have different minimum and maximum voltage levels as well as different phases while charging/discharging.

> The BMS is responsible to measure the voltage (and/or other indicators) and decide, in which state of charge (SOC) the cells are in order to apply the correct current and/or voltage.

Reviewer: Which are the other indicators? Probably you mean current and temperature? Should be more specific.

Also the SOC cannot be measured directly, but has to be calculated by the BMS.

It will optimise the charging rate and determine when to stop charging/discharging.

> To disconnect the battery from the load/charger, different types of switches can be used. For low current systems, a MOSFET switch is often easiest to use, while a mechanical or solid state relais can be necessary to switch higher currents.

Reviewer: Main issue with MOSFETs is that they exist for a limited voltage range only. That's why high-voltage batteries (for electric vehicles) use relays.

### Thermal Protection

> In order to keep the battery in a safe operational state, the thermal managmenent is vital. The system can either be active or passive, using mostly air or another liquid coolant. The necessary hardware like fans and their electronics are mostly consider part of the BMS as well.

Reviewer: I would not consider fans etc. as part of the BMS. They are part of the battery pack, but controlled by the BMS.

In our applications we will only have passive cooling. That should be the case also for larger stationary systems, as power levels in stationary systems are not very high compared to automotive batteries.

### Balancing

> To optimize performance and lifetime of battery packs, balancing is used to distribute load and prevent localized over- or under-charging of cells. Since cells differ slightly in capacity depending on the quality of the manufacturing process and used materials or by cell-aging, cells connected in series can show different voltages. To normalize the SOC for each cell, a BMS can either actively transfer energy from higher charged cells to those with lower SOC or passivly limiting the charging, mostly by wasting energy from cells with 100% SOC until every cell is fully charged. 

Reviewer: The charging is not really limited in passive balancing systems, as the balancing currents are really low compared to charging currents. So the balancing happens after charging so that they get to an equal SOC in the long term.

> Passive balancing can be done by bleeding energy to ground through a resistor and a transistor or bypassing the cell with an appropriate resistor, dropping the voltage across. In any case, passive balancing increases the thermal energy that needs to be dissipated by the thermal protection and decreases the overall efficieny.

Reviewer: Not sure if "bypassing" is the right word. I would associate bypassing more with a diode. The balancing resistor is connected in parallel to the cell and discharges it (like a normal consumer would do).
Also the thermal energy does not go into the cell so it is not really related to the thermal protection mechanisms of the battery. You only have to take care that the PCB with the balancing resistors can dissipate enough heat.

Active balancing aims to distribute the energy better while charging and discharging, but need much more circuitry. Examples....

### Misc

A BMS can also feature communication capabilities to distribute information to other BMS or [charge-controllers](charge_controller.md) or logging systems. Commonly used for standalone setups is a serial connection while BUS connections like CAN are preferably used in systems with several components. See chapter [Communication](../development/communication.md) for more.

> A common feature may also be the connection to a load, including a precharge system to mitigate spikes in current or voltage when the load is connected.

Reviewer: Can you please split this into "Communication" and "Bus precharge" instead of "Misc" and explain the precharging function in a bit more detail? (it slowly charges the large input capacitors of connected power electronics like inverters in order to prevent high inrush currents, see e.g. section 11.5.2 of [bq76952 datasheet](https://www.ti.com/lit/ds/symlink/bq76952.pdf?ts=1618902728801&ref_url=https%253A%252F%252Fwww.ti.com%252Fproduct%252FBQ76952))

## Topologies

### Centralised

> One single BMS is connected to every single cell of the battery, monitoring several states in parallel. For standalone setups, where size is fixed and cells are rather uniform, this topology is easiest to implement. The systems developed by LibreSolar follow this centralized approach.

Reviewer: There is only a single BMS per battery, so it's **the** BMS. Suggestion to make this more clear:
```suggestion
The BMS is directly connected to every single cell of the battery, monitoring several states in parallel. For standalone setups, where size is fixed and cells are rather uniform, this topology is easiest to implement. The systems developed by LibreSolar follow this centralized approach.
```

<fig-caption src="system/bms_centralized.svg" caption="Centralized BMS layout" num="1" />

### Distributed

A monitoring unit is connected to each cell, reporting information about the cell to a central controller. A good OpenSource Project using this approach is Stuart Pittaways [diyBMS](https://github.com/stuartpittaway/diyBMSv4).

<fig-caption src="system/bms_distributed.svg" caption="Distributed BMS layout" num="2" />

### Modular

> A modular setup is designed for larger battery packs. A specialized board is connected to 8 to 16 cells, measuring necessary data. Each of those units is than connected to a main BMS controller, who is responsible for the analysis of the data and has to take the appropriate steps. An OpenSource Project can be found [here](https://foxbms.org/).

Reviewer: ```suggestion
A modular setup is designed for larger battery packs. A specialized board is connected to 8 to 16 cells, measuring necessary data. Each of those units is than connected to a main BMS controller, which is responsible for the analysis of the data and has to take the appropriate steps. An Open Source Project can be found [here](https://foxbms.org/).
```

Which are "the appropriate steps". Be more specific or leave it out.

Passive balancing can be done by bleeding energy to ground through a resistor and a transistor or with a resistor parallel to the cell which acts like a small consumer. Passive balancing increases the thermal energy dissipated by the pack and decreases the overall efficieny.

Active balancing aims to distribute the energy better while charging and discharging, but need much more circuitry.

### Communication

A BMS can also feature communication capabilities to distribute information to other BMS or [charge-controllers](charge_controller.md) or logging systems. Commonly used for standalone setups is a serial connection while BUS connections like CAN are preferably used in systems with several components. See chapter [Communication](../development/communication.md) for more.

### Bus precharge

> A common feature may also be the connection to a load, including a precharge system. Precharging protects the battery against high inrush currents when a load is connected. To slowly charge large capacities in the load circuit, the load is connected over a resistor for a short period of time. Afterwards, the resistor is removed and the load connected normally. 

Reviewer: The inrush current goes into the capacitors, not the battery. From battery perspective it would be outrush currents, but that term is not used. And it protects the capacitors, not the battery.

## Topologies

### Centralised

> The BMS is directly connected to every single cell of the battery, monitoring several states in parallel. For standalone setups, where size is fixed and cells are rather uniform, this topology is easiest to implement. The systems developed by LibreSolar follow this centralized approach.

Reviewer: What is meant by a "uniform" cell?

And please add white space between Libre and Solar.

<fig-caption src="system/bms_centralized.svg" caption="Centralized BMS layout" num="1" />

### Distributed

> A monitoring unit is connected to each cell, reporting information about the cell to a central controller. A good OpenSource Project using this approach is Stuart Pittaways [diyBMS](https://github.com/stuartpittaway/diyBMSv4).

Reviewer: Open Source with white space.

<fig-caption src="system/bms_distributed.svg" caption="Distributed BMS layout" num="2" />

### Modular

> A modular setup is designed for larger battery packs. A specialized board is connected to 8 to 16 cells, measuring necessary data. Each of those units is than connected to a main BMS controller, which is responsible for the analysis of the data. An Open Source Project can be found [here](https://foxbms.org/).

Reviewer: "cell voltages and temperatures" instead of "necessary data".
